src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java

```diff
@@ -75,21 +78,24 @@ protected String getCampaignName() {
      * @param campaignName value.
      */
     private void selectCampaign(final String campaignName) {
-        // Switch to new window opened
-        webDriverWait.until(ExpectedConditions.numberOfWindowsToBe(2));
-        Set<String> handles = webDriver.getWindowHandles();
-        for (String winHandle : handles) {
-            if (!parentHandle.equals(winHandle)) {
-                webDriver.switchTo().window(winHandle);
-                break;
-            }
+        parentHandle = webDriver.getWindowHandle();
+        try {
+            UtilSalesforce.switchToNewWindow(parentHandle);
+            webDriverWait.until(ExpectedConditions.frameToBeAvailableAndSwitchToIt(resultsFrame));
+            clickCampaignSelected(campaignName);
+        } finally {
+            webDriver.switchTo().window(parentHandle);
         }
-        // Perform the actions on new window
+    }
+
+    /**
+     * Clicks to campaign selected.
+     * @param campaignName string value.
+     */
+    private void clickCampaignSelected(final String campaignName) {
```

Comment:
click**On**C

Reply:
it is passive, i will change OnCampaignSelected